File: Main/js/test1.js
const url = "http://localhost:8000/generate_study_plan";

// Data to be sent in the POST request
const data = {
    topics: ["Flutter Fundamentals", "Flutter for app development"],
    main_topic: "Flutter for Beginners"
};

// Function to send the POST request
async function sendPostRequest() {
    try {
        // Send POST request
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(data)
        });
        
        // Check if the request was successful
        if (response.ok) {
            // Parse the JSON response
            const studyPlan = await response.json();
            
            // Access the study_plan object directly
            const plan = studyPlan.study_plan;
            
            console.log("Study Plan:");
            console.log(plan);
            document.getElementById("API-info").innerHTML = plan;
        } else {
            console.error(`Error: ${response.status}`);
            console.error(await response.text());
        }
    
    } catch (error) {
        console.error(`An error occurred: ${error}`);
    }
}

// Run the function
sendPostRequest();

File: Hackathon copy/test1.py
import requests
import json

# URL of your FastAPI server
url = "http://localhost:8000/generate_study_plan"

# Data to be sent in the POST request
data = {
    "topics": ["Flutter Fundamentals", "Flutter for app development"],
    "main_topic": "Flutter for Beginners"
}

# Headers
headers = {
    "Content-Type": "application/json"
}

def send_post_request():
    try:
        # Send POST request
        response = requests.post(url, data=json.dumps(data), headers=headers)
        
        # Check if the request was successful
        if response.status_code == 200:
            # Parse the JSON response
            study_plan = response.json()
            
            print("Study Plan:")
            print(json.dumps(study_plan, indent=2))
        else:
            print(f"Error: {response.status_code}")
            print(response.text)
    
    except requests.exceptions.RequestException as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    send_post_request()